ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/fees/CoinbaseFeePriceCalculator.java

```diff
@@ -22,16 +22,17 @@
 
 @FunctionalInterface
 public interface CoinbaseFeePriceCalculator {
-  Wei price(Gas coinbaseFee, Wei transactionGasPrice, Optional<Long> baseFee);
+  long price(Gas coinbaseFee, Wei transactionGasPrice, Optional<Long> baseFee);
```

Comment:
why not return Wei?

Reply:
Because Wei does not accept negative value. It is not worth accepting it just for this specific use case in my opinion.